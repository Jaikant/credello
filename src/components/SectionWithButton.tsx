import * as React from 'react';
import styled from 'styled-components';
import { device } from 'libs/device';
import Button from '@material-ui/core/Button';
import { TextBold, Text } from 'components/Typography';

const SectionWrapper = styled.div`
  display: grid;
  grid-gap: 16px;
  padding: 32px 16px;
  color: #fff;
  background-color: #5cdb95;
  margin: 16px -16px;
  @media ${device.tablet} {
    grid-gap: 32px;
    margin: 16px -24px;
    padding: 32px 24px;
    justify-content: center;
  }
  @media ${device.laptop} {
    grid-gap: 32px;
    margin: 16px -128px;
    padding: 32px 128px;
    justify-content: center;
  }
`;

const StyledButton = styled(Button)`
  background: #fff;
  &.MuiButton-contained {
    color: #5cdb95;
    padding: 16px;
    background: #fff;
  }
  @media ${device.tablet} {
    max-width: 400px;
  }
  @media ${device.laptop} {
    max-width: 400px;
  }
`;

// type Props = {
// };

const SectionWithButton = () => {
  return (
    <SectionWrapper>
      <TextBold size="24px">Take control of your finances</TextBold>
      <Text>
        We analyze all of your options, so you can act with confidence
      </Text>
      <StyledButton variant="contained" disableElevation>
        <TextBold size="14px">Manage Debt</TextBold>
      </StyledButton>
    </SectionWrapper>
  );
};

SectionWithButton.defaultProps = {};

export default SectionWithButton;
