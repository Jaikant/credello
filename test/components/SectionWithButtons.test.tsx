import React from 'react';
import { render } from '../testUtils';
import SectionWithButton from '../../src/components/SectionWithButton';
import 'jest-styled-components';

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<SectionWithButton />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
