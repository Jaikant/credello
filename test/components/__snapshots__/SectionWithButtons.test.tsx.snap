// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  .c2 {
  font-family: Roboto;
  font-weight: 500;
  font-size: 16px;
  line-height: 1.2;
  margin: 0;
  margin-bottom: 0;
  margin-top: 0;
  color: inherit;
  text-align: initial;
}

.c1 {
  font-family: Roboto;
  font-weight: 700;
  font-size: 24px;
  line-height: 1.2;
  margin: 0;
  margin-bottom: 0;
  margin-top: 0;
  color: inherit;
  text-align: initial;
}

.c4 {
  font-family: Roboto;
  font-weight: 700;
  font-size: 14px;
  line-height: 1.2;
  margin: 0;
  margin-bottom: 0;
  margin-top: 0;
  color: inherit;
  text-align: initial;
}

.c0 {
  display: grid;
  grid-gap: 16px;
  padding: 32px 16px;
  color: #fff;
  background-color: #5cdb95;
  margin: 0px -16px;
}

.c3 {
  background: #fff;
}

.c3.MuiButton-contained {
  color: #5cdb95;
  padding: 16px;
  background: #fff;
}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {
  .c1 {
    font-size: 24px;
  }
}

@media (min-width:1224px) {
  .c1 {
    line-height: 1.5;
    font-size: 24px;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 14px;
  }
}

@media (min-width:1224px) {
  .c4 {
    line-height: 1.5;
    font-size: 14px;
  }
}

@media (min-width:768px) {
  .c0 {
    grid-gap: 32px;
    margin: 0px -24px;
    padding: 32px 24px;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (min-width:1224px) {
  .c0 {
    grid-gap: 32px;
    margin: 0px -128px;
    padding: 32px 128px;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (min-width:768px) {
  .c3 {
    max-width: 400px;
  }
}

@media (min-width:1224px) {
  .c3 {
    max-width: 400px;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      Take control of your finances
    </div>
    <div
      class="c2"
    >
      We analyze all of your options, so you can act with confidence
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained c3 MuiButton-disableElevation"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <div
          class="c4"
        >
          Manage Debt
        </div>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
